docs/cheat-sheets/java-xxe.mdx

```diff
@@ -0,0 +1,1839 @@
+For various classes, this feature can protect the parser from all available payloads. However, this is not the case for all of the classes that have been tested to create this cheat sheet.
+
+#### Disabling DTD processing
+
+To disable DTD processing with one of the available mechanisms:
+
+- The `setFeature` method with the following URL: `http://apache.org/xml/features/disallow-doctype-decl`
+- The `setProperty` method to set `XMLInputFactory.SUPPORT_DTD` to false.
+- The `setAttribute` method to set `XMLConstants.ACCESS_EXTERNAL_DTD` to an empty string (`“”`). 
+
+For some parsers, there are multiple methods to disable DTD processing. For most parsers, this setting adds additional security for processing XML documents. However, this does not affect the processing of schemas or stylesheets. Still, there are classes such as `DocumentBuilderFactory`, `Validator`, and `SchemaFactory` with multiple methods to disable DTD, and setting the above-mentioned features throws no exceptions, but only one of them has any effect on the security of the parser.
+
+#### Disabling external schema and stylesheet processing
+
+The `setAttribute` method can be used to set `XMLConstants.ACCESS_EXTERNAL_SCHEMA` and `XMLConstants.ACCESS_EXTERNAL_STYLESHEET` to an empty string to disable these features.
+The effects on security, however, are often limited with no observed effect for several classes.
+
+#### Disabling external entities
+
+Set `setFeature` to `false` to disable `http://xml.org/sax/features/external-general-entities` and `http://xml.org/sax/features/external-parameter-entities` features. When both of these features are disabled, all parsers are protected against all of the tested payloads.
+Except for the `Validator` class, where they have no effect at all.
+There is also `setExpandEntityReferences` that can be set to `false` and seems to have the same effect as disabling external general entities, based on our testing.
+
+## 3. Overview of the effect of security measures for each class
+
+Summarizing security measures against XXE is not easy as not all security measures work for every class (as mentioned in section [2.B Security measures](#2b-security-measures)). The tables below this section can help you to choose the best security measures against a specific vulnerability.
+
+:::info Table legend
+- Each table represents one Java class. For example `DocumentBuilderFactory`.
+- Columns display attack payloads that can be potentially used to exploit a vulnerability (for example DTD or XML bomb).
+- Rows represent ways to configure the parsers.
+
+✅ - Secure<br />
+❌ - Not Secure<br />
+⚠️  - An exception is thrown<br />
+N/A - Not available
+:::
+
+### 3.A DocumentBuilderFactory
+
+[`javax.xml.parsers.DocumentBuilderFactory`](https://docs.oracle.com/javase/7/docs/api/javax/xml/parsers/DocumentBuilderFactory.html) can be used to process all payload types. By default it is vulnerable against 4 of the attack payloads.
```

Comment:
I noticed that all of the Oracle links point to the Java 7 docs, which hit EOL last year. Is there a more recent version of the docs that you'd feel comfortable pointing to? Hopefully there isn't much of a diff between 7 and 19 (latest version)

Reply:
Good catch!